include/soci/rowset.h

```diff
@@ -148,8 +148,8 @@ class rowset_impl
 
     unsigned int refs_;
 
-    const cxx_details::auto_ptr<statement> st_;
-    const cxx_details::auto_ptr<T> define_;
+    const std::unique_ptr<statement> st_;
+    const std::unique_ptr<T> define_;
     SOCI_NOT_COPYABLE(rowset_impl)
```

Comment:
This is no longer needed as the `unique_ptr` members prevent copying anyway.
Dunno if you want to keep this around to make it more explicit, though

Reply:
I haven't decided what to do about `SOCI_NOT_COPYABLE` yet, but I'm probably going to remove it and declare the special member functions directly. It doesn't need to be done as part of this PR however.